stylegan_code_finder/train_with_lightning.py

```diff
@@ -69,45 +51,39 @@ def main(rank: int, args: argparse.Namespace, world_size: int):
     else:
         val_data_loader = None
 
-    train_builder_class = get_train_builder_class(config)
-    training_builder = train_builder_class(config, train_data_loader, val_data_loader, rank=rank, world_size=world_size)
-
-    pathlib.Path(args.log_dir, args.log_name).mkdir(parents=True, exist_ok=True)
+    config['num_iter_epoch'] = len(train_data_loader)
+    Path(args.log_dir, args.log_name).mkdir(parents=True, exist_ok=True)
 
     logging.info("Initializing wandb... ")
     logger = WandbLogger(name=args.log_name, save_dir=args.log_dir, project=args.wandb_project_name)
     logging.info("done")
 
-    plot_data_loader = val_data_loader
-    if val_data_loader is None:
-        plot_data_loader = train_data_loader
-    segmentation_plotter = SegmentationPlotter(plot_data_loader, config)
     segmenter_class = get_segmenter_class(config)
-    segmenter = segmenter_class(training_builder, config, segmentation_plotter, logger)
-
-    #snapshotter = training_builder.get_snapshotter()
-    #if snapshotter is not None:
-    #    trainer.extend(snapshotter)
+    segmenter = segmenter_class(config)
 
-    #image_plotter = training_builder.get_image_plotter()
-    #if image_plotter is not None:
-    #    trainer.extend(image_plotter)
-
-    if 'max_iter' in config:
-        config['epochs'] = 1000
+    if config['network'] == 'EMANet':
+        multiprocessing_strategy = 'ddp'
     else:
-        config['max_iter'] = -1
+        multiprocessing_strategy = 'ddp_find_unused_parameters_false'
+
+    model_checkpoint_callback = ModelCheckpoint(save_top_k=10, monitor='val_loss',
+                                                dirpath=Path(config['log_dir'], 'checkpoints'),
+                                                filename="segmentation-{epoch:02d}-{val_loss:.2f}")
```

Comment:
Top k=10 sounds like a lot. I would argue that 2/3 are enough (this option is to save disk space and writing operations after all).